### DISCUSSION TITLE
Android 13 to receive "app streaming" ability, maybe a nice way to add remote control to the Android agent

### DISCUSSION THREAD

**Discussion by veitw** [Ylianst/MeshCentral#3652] [Ideas]
Hi @Ylianst,

I just read [this article at 9to5google.com](https://9to5google.com/2022/02/14/exclusive-pixel-stream-android-apps-chromebook-pc-video/) about a new feature to arrive in Android 13; the ability to stream apps to desktop devices via a Web App.

Maybe this could be used to add real management capabilities to the Android agent, though it would be background management, which would be both a pro as to not bother the device user, but also a con as it still does not allow to show the device user how to do things.

Best regards,
// Veit

**Comment by theoneandonly-vector** [comment]
there is an opensource app which can do this without android 13 already:
https://github.com/h-mdm/remote-control-android
https://github.com/h-mdm/remote-control

**Reply by veitw** [reply]
Nice, I did not know this one so far. It is not exactly the same, as it shares the screen, but for most use cases, this one might be even better.

@Ylianst: Maybe have a look at this one for the remote screen control capabilities employing accessibility feature? This reads like a nice way to control the device without USB debugging enabled or rooting.

**Reply by Ylianst** [reply]
Agreed, this is interesting. Right now, the Android agent should run on Android 6 or higher, but if there are ways to inject input back to the Android device, that would be excellent.

**Reply by veitw** [reply]
I also like screen overlay this projects shows on the screen, which could be used as an equivalent to the privacy bar on desktop machines.

**Reply by LordBritish2000** [reply]
Cool.. Their gesturehandling without root or hardwareoptimization:
https://github.com/h-mdm/remote-control-android/blob/master/app/src/main/java/com/hmdm/control/GestureDispatchService.java

**Reply by rallisf1** [reply]
I also found https://github.com/Genymobile/scrcpy which uses ADB but I'm not sure it will be required for the MC agent.

**Comment by veitw** [comment]
@Ylianst : I stumbled over another open-source remote management system called RustDesk, implemented in Rust. It follows a more TeamViewer-/Anydesk-like approach than MeshCentral, but they managed to have input on Android without requiring root, by registering the app as an input method. The user still has to enable permission for this.

https://rustdesk.com/docs/en/manual/control-mobile/

**Reply by LordBritish2000** [reply]
Cool. I tried rustdesk once. It works good, but the Server is not really open source.

Android mouse and keyboard input would be really great :-) :-) :-)
